pandas/io/excel/_odfreader.py

```diff
@@ -182,6 +183,20 @@ def _get_column_repeat(self, cell) -> int:
 
         return int(cell.attributes.get((TABLENS, "number-columns-repeated"), 1))
 
+    def _parse_odf_time(self, value: str) -> pd.Timestamp:
+        """
+        Helper function to convert ODF variant of ISO 8601 formatted duration
+        "PnYnMnDTnHnMnS" - see https://www.w3.org/TR/xmlschema-2/#duration
+        """
+        parts = re.match(r"^\s*PT\s*(\d+)\s*H\s*(\d+)\s*M\s*(\d+(\.\d+)?)\s*S$", value)
+        if parts is None:
+            raise ValueError(f"Failed to parse ODF time value: {value}")
```

Comment:
Can you add a test that hits this?

Reply:
Sure, on 6a5d4eb8d19dbbbc11811dbdc3a2792a8f5690e3 I've added new sample with manually corrupted time value.